pyccel/ast/numpyext.py

```diff
@@ -534,12 +614,12 @@ def __init__(self, shape, fill_value, dtype=None, order='C'):
         order = NumpyNewArray._process_order(order)
 
         # Cast fill_value to correct type
-        # TODO [YG, 09.11.2020]: treat difficult case of LiteralComplex
         from pyccel.ast.datatypes import str_dtype
         stype = str_dtype(dtype)
-        if fill_value and stype != 'complex':
+        if fill_value:
             from pyccel.codegen.printing.fcode import python_builtin_datatypes
             cast_func  = python_builtin_datatypes[stype]
+            cast_func = process_stype(stype, precision)
```

Comment:
Why do we need to use `stype`? Can't we use the `dtype` directly?

Reply:
i think `stype` is enough because we need just a dictionary that is more flexible than `python_builtin_datatypes`.

Reply:
Can the Native types not be dictionary keys?

Reply:
we can do that, by `_name` attribute in type objects.

Reply:
What I meant was:
```
PrecisionToDtype = {
    NativeInteger() : {
        4 : NumpyInt32,
...
```
But using the `_name` is good too :+1: